desc/optimize/utils.py

```diff
@@ -220,27 +357,29 @@ def print_iteration_nonlinear(
     if cost is None or abs(cost) == np.inf:
         cost = " " * 15
     else:
-        cost = "{:^15.4e}".format(cost)
+        cost = "{: ^15.3e}".format(cost)
```

Comment:
its part of the format specifier: https://docs.python.org/3/library/string.html#format-specification-mini-language

Reply:
This is just the fill character? It sounds like it defaults to a space anyways. My point is that we should be consistent: lines 360 and 365 explicitly ask for a space, while lines 370 and 375 (and everywhere else in the code, I think) omit it (and I think they give the same behavior.

Reply:
this tells it to print a space or a minus sign. It's really only an issue for things that can be negative (most things we print aren't, but with the augmented lagrangian the cost and delta cost can be) and it's also only an issue when we want to align things vertically.

ie, as we had it previously, things would look like:
```
Cost reduction
1.234
-1.234
2.234
-4.567
``` 

while this change makes it looks like
```
Cost reduction
 1.234
-1.234
 2.234
-4.567
``` 

so that things are lined up all pretty